ideas/134-seamless-login.md

```diff
@@ -0,0 +1,91 @@
+<!-- in PR the document should be named as`DEV#1-title.md` -->
+
+## Preamble
+
+    Idea: 132
+    Title: Seamless Login
+    Status: Draft
+    Created: 2018-05-04
+
+
+
+## Summary
+<!-- "If you can't explain it simply, you don't understand it well enough." Provide a simplified and layman-accessible explanation of the Idea. -->
+
+When a user opens Status or taps on a push notification they should be able to start using the app quickly without any interruptions, and also have control over authentication preferences.
+
+## Swarm Participants
+<!-- Each contributor pledges to the idea with their FOCUS value. (hours per week) -->
+<!-- Here all roles in swarm are defined and filled, one of the contributors should responsibility of the Idea as Lead. -->
+
+<!-- Testing/Evaluation support role is also mandatory to check in on specified Goal dates or earlier. -->
+
+<!-- Lead Contributor is the Owner of the Idea. If required, they can get support from a PM, but should be responsible for end to end execution of the Idea. This includes ensuring appropriate resources are allocated, setting realistic timelines and milestones, and any post-launch metrics or bug fixes that are attributed to the Idea -->
+<!-- A swarm requires at minimum 3 contributors and 1 evaluator/tester -->
+<!-- 'Contributor' should be replaced with a descriptive role type. -->
+- Lead Contributor: @alwx
+- Testing & Evaluation: @asemiankevich
+- PM: @chadyj
+- UX: @andmironov @jpbowen @hesterbruikman
+
+
+## Product Overview
+<!-- A short (~200 word) description and motivation of the Idea. Without clear explanation the Idea should not proceed. Can include User Stories -->
+<!-- Testing/Evaluation role accepts responsbility to checkin at Goal dates, -->
+<!-- forces discussion to continue implementation or recommend disband and post-mortem. -->
+
+Status frequently logs out users when the app is in the background and forces users to log in again when opening the app. Being repeatedly and unexpectedly logged out has led to frustration and complaints (see Instabug and internal reports) which negatively impact the user experience and potentially contributes to churn. Also, users are encouraged to have complex passwords but this can be cumbersome to enter.
+
+By reducing friction around the critical point of returning to the app users can spend more time in Status which will help DAU and retention (Q2 OKR).
+
+This idea seeks to make logging in seamless through both technical and UX methods.
+
+
+### Product Description
+<!-- What functionality are you adding? What will this look like from a user perspective? Why is this important? -->
+
+From a users perspective, the user will not repeatedly be logged out when using Status. This will be the default behavior. Since authentication preference is personal, there will be security/login settings so that a user can customize their desired level of control. For example, a user can specify never to be logged out, or choose to log in every time.
```

Comment:
This came up in Slack and it has significant security impact, since Whisper keypair is used to chat: https://status-im.slack.com/archives/C8HGM58AU/p1522678563000639?thread_ts=1522668853.000146&cid=C8HGM58AU

So while this idea is great, the very first step should be to understand if this is even technically feasible to do, without compromising core security.

Reply:
This is what we're going. We're discussing things and making sure it's technically feasible.